Sources/NSTextCheckResult+NamedCapture.swift

```diff
@@ -0,0 +1,54 @@
+//
+//  NSTextCheckResult+NamedCapture.swift
+//  NSRegExNamedCaptureGroup
+//
+//  Created by Rock Yang on 2017/6/24.
+//
+//
+
+import Foundation
+import ObjectiveC
+
+
+extension NSTextCheckingResult {
+  
+  /// Returns the result type that the range represents.
+  /// A result must have at least one range, but may
+  /// optionally have more (for example, to represent regular
+  /// expression capture groups).
+  ///
+  /// @pram groupName The name of capture group that appears in the regex
+  ///       pattern. Passing the value `nil` if the overall range is expected.
+  ///
+  /// @return The range of the result.
+  ///         Passing the method the value `nil` always returns
+  ///         the value of the the `range` property. Additional ranges,
+  ///         if any, can be retrieved through their capture group names.
```

Comment:
Our sole public API is Swifty 🐣 now, so I suppose the documentation markup should also be Swift-style.